client/dogs/SearchPage.js

```diff
@@ -85,35 +74,31 @@ export default class IndexPage extends React.Component {
   };
 
   handleSubmit = (ev: SyntheticInputEvent) => {
```

Comment:
I'd remove this; doing a `toString` on the `state` object is not the right way to build up query parameters. This isn't called now anyway since you're doing the normal form GET method.